optimizely/client/factory.go

```diff
@@ -43,6 +42,47 @@ type OptimizelyFactory struct {
 	Datafile []byte
 }
 
+// OptionFunc is a type to a proper func
+type OptionFunc func(*OptimizelyClient, utils.ExecutionCtx)
+
+// GetClient gets client and sets some parameters
+func (f OptimizelyFactory) GetClient(clientOptions ...OptionFunc) *OptimizelyClient {
+	executionCtx := utils.NewCancelableExecutionCtx()
+	appClient := &OptimizelyClient{
+		isValid:      false,
```

Comment:
This should be true right? Otherwise the client is unusable

Reply:
I see, the question is what are the cases that make the client invalid ? What is the minimum requirement to make the client valid? I don't want to set it blindly to true, otherwise we don't need that variable at all.